lib/fs.js

```diff
@@ -1615,6 +1615,11 @@ function ReadStream(path, options) {
   if (!(this instanceof ReadStream))
     return new ReadStream(path, options);
 
+  if (typeof options === 'string')
+    options = { encoding: options };
+  else if (options && typeof options !== 'object')
```

Comment:
You probably should be more strict with this check. For example, `options = 0` would slip through.